lms/djangoapps/verify_student/views.py

```diff
@@ -645,15 +656,16 @@ def create_order(request):
         donation_for_course[unicode(course_id)] = amount
         request.session['donation_for_course'] = donation_for_course
 
-    # prefer professional mode over verified_mode
-    current_mode = CourseMode.verified_mode_for_course(course_id)
+    no_id_prof_mode = CourseMode.mode_for_course(course_id, CourseMode.NO_ID_PROFESSIONAL_MODES[0])
+    # prefer NO-ID-PROFESSIONAL mode over professional mode then verified_mode
+    current_mode = no_id_prof_mode if no_id_prof_mode else CourseMode.verified_mode_for_course(course_id)
 
-    # make sure this course has a verified mode
-    if not current_mode:
+    # make sure this course has a verified mode if it is not no_id_professional mode
+    if not no_id_prof_mode and not current_mode:
         log.warn(u"Verification requested for course {course_id} without a verified mode.".format(course_id=course_id))
         return HttpResponseBadRequest(_("This course doesn't support verified certificates"))
```

Comment:
I think we can safely remove this check now.  The flow will need to support both verified and non-verified course modes, so we shouldn't need to return a 400 :)